python/ray/train/tests/test_trainer_restore.py

```diff
@@ -392,7 +392,7 @@ def check_for_raise():
                 trainer_cls.restore(str(tmpdir))
 
         if should_warn:
-            with pytest.warns() as warn_record:
+            with pytest.warns(Warning) as warn_record:
```

Comment:
There's a bunch of tests in this file that use a stopper instead to finish after the originally specified boosting rounds. We should remove those stopping conditions and test the new logic.

Check `test_gbdt_trainer_restore`:

```python
        run_config=RunConfig(
            local_dir=str(tmpdir),
            name=exp_name,
            checkpoint_config=CheckpointConfig(num_to_keep=1, checkpoint_frequency=1),
            callbacks=[FailureInjectionCallback(num_iters=2)],
            # We also use a stopper, since the restored run will go for
            # another 5 boosting rounds otherwise.
            stop={"training_iteration": 5},
        ),
```

Reply:
Got it, thanks. Odd that it passes though!